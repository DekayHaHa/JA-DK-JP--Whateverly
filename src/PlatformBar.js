import React, { Component } from 'react';
import CheckBox from './CheckBox.js';

export default class PlatformBar extends Component {
  constructor(props) {
    super(props);
    this.state = {showPlatforms: false};
  }
  togglePlatforms = () => {
    this.state.showPlatforms ? 
      (this.setState({ showPlatforms: false })) : 
      (this.setState({ showPlatforms: true }));
    }
  render() {
    let hidePlatforms = "";
    this.state.showPlatforms ? 
      (hidePlatforms = "platform-check") : 
      (hidePlatforms = "platform-check hide");
    let platforms = this.props.platforms.sort();
    return (
      <div className="console-bar search-section">
          <form className={hidePlatforms}>
            {
              platforms.map((val, i) => {
                return <CheckBox name={val}
                  updateChecks= {this.props.updateChecks}
                  class="platformsToFilter check-box"
                  key={i} />
              })
            }
          </form>
        <button onClick={this.togglePlatforms} className="button">PLATFORM</button>
      </div>
    )
  }
}
