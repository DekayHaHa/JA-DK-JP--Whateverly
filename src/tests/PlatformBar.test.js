import React from 'react';
import ReactDOM from 'react-dom';
import PlatformBar from '../PlatformBar';
import { shallow } from 'enzyme';

let plats = ['PS4', 'PS3', 'XBOX']
describe('PlatformBar', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <PlatformBar platforms={plats}/>
    );
  });
  it('should match snapshop', () => {
    expect(wrapper).toMatchSnapshot();
  });
})