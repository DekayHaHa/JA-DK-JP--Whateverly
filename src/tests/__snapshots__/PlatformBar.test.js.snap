// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlatformBar should match snapshop 1`] = `
<div
  className="console-bar search-section"
>
  <form
    className="platform-check"
    onChange={[Function]}
  >
    <Footer
      class="platform-check-box"
      key="0"
      name="PS4"
    />
    <Footer
      class="platform-check-box"
      key="1"
      name="PS3"
    />
    <Footer
      class="platform-check-box"
      key="2"
      name="XBOX"
    />
  </form>
  <button
    className="button"
  >
    PLATFORM
  </button>
</div>
`;
